### PR TITLE
29 Fix NA buyer/seller name index error

### LINKED ISSUE #29: Condo flagging bug
When running flagging on pre-2017 data, we get an `IndexError` in the `name_selector()` function. It could have something to do with the NA values in buyer and seller name.

### PR DESCRIPTION
**Pull request by wagnerlmichael**
The code now runs with the entire 2014- sales query.  I believe the reason that it was not running was because `name_selector()` didn't handle the case where `split_logic()` looked at NA values in buyer or seller name.

`name_selector()` takes the return value of `split_logic()` as an argument in `clean_id()`. Specifically this happens in these two lines:

``` 
df["sv_buyer_id"] = df.apply(clean_id, args=("sv_buyer",), axis=1)
df["sv_seller_id"] = df.apply(clean_id, args=("sv_seller",), axis=1)
```
The fix here essentially converts the NA values to an "Empty Name" string value.

Before, there was an out of index error, since the NA value was being indexed.

Closes #29

### CODE REVIEW

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (dfsnow) [inline glue/flagging_script_glue/flagging_e894e5.py:1099]: **question (non-blocking):** Shouldn't this line also be `elif` given the proceeding line? Think carefully about the control flow here.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:551]: This is changed because we have standardized pandas versions in glue and on local.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:897]: The prior flagging script was erroneously matching values with different string representations of NA. Since there are null values, the `words = str(row[column]).lower()` directly above this code chunk was converting those to `none` or `nan`, which returned family sales matches. 

There were also other already present values such as `UNKOWN`, `MISSING SELLER NAME`, and `MISSING BUYER NAME`, which were returning matches. We decide here to manually change these values to `Empty Name`, using the existing strategy.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:1226]: This was an unused function which was an exact duplicate of the `df_name_match` function.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:1076]: (No real names used in this comment) This small regex fix was what actually fixed the error. After this regex, there is a split on the match which take the left element. In one case a name had the structure like `A. K. allison`, so it split on this name returning an empty string. The old match structure also hit names like:

- Samantha K. Adams
- Augustina K. Anthony

We don't want these matches, so I added a lookahead to see if there was another letter after the final `'a'`. After manually reviewing results, this improved matching.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:1101]: This change prevents the same error from occurring.

In order to make sure the index error from returning an empty list doesn't cause the whole job to break, I changed this such that if it happens again, it will just read it as `'Empty Name'`, instead of a null list value.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [inline glue/flagging_script_glue/flagging_e894e5.py:1105]: We simplify the logic here as the prior code in this function was redundant (grabbing last element of list)

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (dfsnow) [reply glue/flagging_script_glue/flagging_e894e5.py:897]: **praise:** Nice catch @wagnerlmichael.

**Question:** I assume that the name matching down the line discards any matches on `"Empty Name"`, is that correct?

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (dfsnow) [reply glue/flagging_script_glue/flagging_e894e5.py:1076]: **praise:** Nice find again @wagnerlmichael. This kind of stuff is super tricky to debug.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (dfsnow) [reply glue/flagging_script_glue/flagging_e894e5.py:1101]: **request:** Can you add more context about what the error is here, so that I (and future parties) can understand exactly what was breaking and what this will fix?

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (dfsnow) [inline glue/flagging_script_glue/flagging_e894e5.py:1103]: **nitpick:** What if the last two elements match one of these? i.e. James Kirk Jr. III.

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [reply glue/flagging_script_glue/flagging_e894e5.py:897]: Yes, that is correct

File: manual_flagging/src/flagging_rolling.py
Comment (wagnerlmichael) [inline manual_flagging/src/flagging_rolling.py:1107]: This fixes the problem of multiple suffixes like `'Bob jr iii'`

File: glue/flagging_script_glue/flagging_e894e5.py
Comment (wagnerlmichael) [reply glue/flagging_script_glue/flagging_e894e5.py:1103]: Sure. So in `split_logic()`, there are various scenarios where text is split on a certain text match. The error that was happening happened to do with a name like `'A. K. Arnold'`. In the old logic, the `'a k a'` was detected as a match, and split on this name. 

In that part of `split_logic()`, we take the content of whatever was to the _left_  of the split. In the case of  `'A. K. Arnold'`, the left part of the string after the split would have been `null`, or an empty string. Interestingly, this never happened until one problematic name in the condo data came through. So the output `split_logic()` was an empty list, in this case.

Then, the output from `split_logic()` feeds into `name_selector()`, the input was an empty list, and the code from name selector attempts to index a value in the empty list, leading to an index error.

The fix then, just manually adjusts the empty list to an `'Empty Name`' value, which will just lead to a non-matched name.

Let me know if this makes sense @dfsnow

Comment (dfsnow) [reply glue/flagging_script_glue/flagging_e894e5.py:1103]: Checks out, thanks for the extra context!

File: manual_flagging/src/flagging_rolling.py
Comment (dfsnow) [reply manual_flagging/src/flagging_rolling.py:1107]: I appreciate the handling for the unfortunate soul named `Jr. Jr. Jr.`.

**Review by dfsnow — APPROVED** [review]
Looks good @wagnerlmichael! I'd ask Jean about merge order here since we might want to wait until the terraform merge is done.

**Pull request merged**
